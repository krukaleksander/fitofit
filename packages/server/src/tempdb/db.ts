export const db = [
  {
    id: 0,
    name: 'Aerobik',
    cal: 550,
  },
  {
    id: 1,
    name: 'Aerobik w wodzie',
    cal: 580,
  },
  {
    id: 2,
    name: 'Badminton',
    cal: 400,
  },
  {
    id: 3,
    name: 'Bieg (km przez 3,40 min)',
    cal: 1200,
  },
  {
    id: 4,
    name: 'Bieg (km przez 5 min)',
    cal: 1000,
  },
  {
    id: 5,
    name: 'Ćwiczenia na siłowni',
    cal: 400,
  },
  {
    id: 6,
    name: 'Chodzenie po sklepie',
    cal: 250,
  },
  {
    id: 7,
    name: 'Czytanie',
    cal: 25,
  },
  {
    id: 8,
    name: 'Czytanie głośno',
    cal: 150,
  },
  {
    id: 9,
    name: 'Golf',
    cal: 250,
  },
  {
    id: 10,
    name: 'Gotowanie',
    cal: 105,
  },
  {
    id: 11,
    name: 'Gra na pianinie 15 min.',
    cal: 120,
  },
  {
    id: 12,
    name: 'Gra w kręgle',
    cal: 300,
  },
  {
    id: 13,
    name: 'Gra w piłkę nożną',
    cal: 650,
  },
  {
    id: 14,
    name: 'Yoga',
    cal: 288,
  },
  {
    id: 15,
    name: 'Intensywna gimnastyka',
    cal: 300,
  },
  {
    id: 16,
    name: 'Jazda konna',
    cal: 650,
  },
  {
    id: 17,
    name: 'Jazda na deskorolce',
    cal: 371,
  },
  {
    id: 18,
    name: 'Jazda na nartach biegowych',
    cal: 574,
  },
  {
    id: 19,
    name: 'Jazda na nartach wodnych',
    cal: 441,
  },
  {
    id: 20,
    name: 'Jazda na nartach zjazdowych',
    cal: 511,
  },
  {
    id: 21,
    name: 'Jazda na rolkach',
    cal: 400,
  },
  {
    id: 22,
    name: 'Jazda na rowerze (10 km/h)',
    cal: 300,
  },
  {
    id: 23,
    name: 'Jazda na rowerze (20 km/h)',
    cal: 600,
  },
  {
    id: 24,
    name: 'Jogging',
    cal: 600,
  },
  {
    id: 25,
    name: 'Kopanie, grabienie',
    cal: 500,
  },
  {
    id: 26,
    name: 'Koszykówka',
    cal: 550,
  },
  {
    id: 27,
    name: 'Lekka gimnastyka',
    cal: 210,
  },
  {
    id: 28,
    name: 'Lekka praca biurowa',
    cal: 140,
  },
  {
    id: 29,
    name: 'Marszobieg',
    cal: 500,
  },
  {
    id: 30,
    name: 'Mycie okien',
    cal: 240,
  },
  {
    id: 31,
    name: 'Mycie podłogi',
    cal: 250,
  },
  {
    id: 32,
    name: 'Odkurzanie',
    cal: 260,
  },
  {
    id: 33,
    name: 'Piłowanie drewna',
    cal: 450,
  },
  {
    id: 34,
    name: 'Ping-pong',
    cal: 280,
  },
  {
    id: 35,
    name: 'Pływanie',
    cal: 400,
  },
  {
    id: 36,
    name: 'Praca ekspedientki',
    cal: 280,
  },
  {
    id: 37,
    name: 'Praca kamieniarska',
    cal: 400,
  },
  {
    id: 38,
    name: 'Praca w ogródku',
    cal: 250,
  },
  {
    id: 39,
    name: 'Pranie ręczne',
    cal: 150,
  },
  {
    id: 40,
    name: 'Prasowanie',
    cal: 144,
  },
  {
    id: 41,
    name: 'Prowadzenie samochodu',
    cal: 126,
  },
  {
    id: 42,
    name: 'Robienie na drutach',
    cal: 55,
  },
  {
    id: 43,
    name: 'Robienie zakupów',
    cal: 300,
  },
  {
    id: 44,
    name: 'Schodzenie ze schodów',
    cal: 364,
  },
  {
    id: 45,
    name: 'Ścieranie kurzy',
    cal: 240,
  },
  {
    id: 46,
    name: 'Siatkówka',
    cal: 450,
  },
  {
    id: 47,
    name: 'Siatkówka plażowa',
    cal: 588,
  },
  {
    id: 48,
    name: 'Skakanie na skakance',
    cal: 574,
  },
  {
    id: 49,
    name: 'Słanie łóżka',
    cal: 200,
  },
  {
    id: 50,
    name: 'Spacer szybkim tempie',
    cal: 300,
  },
  {
    id: 51,
    name: 'Spacer w umiarkowanym tempie',
    cal: 200,
  },
  {
    id: 52,
    name: 'Sen',
    cal: 62,
  },
  {
    id: 53,
    name: 'Śpiew',
    cal: 122,
  },
  {
    id: 54,
    name: 'Sprzątanie łazienki',
    cal: 240,
  },
  {
    id: 55,
    name: 'Sprzątanie pokoju',
    cal: 180,
  },
  {
    id: 56,
    name: 'Squash',
    cal: 497,
  },
  {
    id: 57,
    name: 'Stanie na baczność',
    cal: 115,
  },
  {
    id: 58,
    name: 'Stanie swobodne',
    cal: 100,
  },
  {
    id: 59,
    name: 'Szorowanie podłóg',
    cal: 426,
  },
  {
    id: 60,
    name: 'Szybki marsz 6 km/h',
    cal: 150,
  },
  {
    id: 61,
    name: 'Szycie na maszynie',
    cal: 135,
  },
  {
    id: 62,
    name: 'Szycie ręczne',
    cal: 111,
  },
  {
    id: 63,
    name: 'Taniec w dyskotece',
    cal: 500,
  },
  {
    id: 64,
    name: 'Tenis',
    cal: 450,
  },
  {
    id: 65,
    name: 'Trzepanie dywanów',
    cal: 260,
  },
  {
    id: 66,
    name: 'Ubieranie się i rozbieranie',
    cal: 118,
  },
  {
    id: 67,
    name: 'Układanie dokumentów',
    cal: 200,
  },
  {
    id: 68,
    name: 'Wchodzenie po schodach',
    cal: 1100,
  },
  {
    id: 69,
    name: 'Wędkowanie',
    cal: 300,
  },
  {
    id: 70,
    name: 'Wiosłowanie',
    cal: 500,
  },
  {
    id: 71,
    name: 'Zamiatanie podłogi',
    cal: 100,
  },
  {
    id: 72,
    name: 'Zmywanie naczyń',
    cal: 114,
  },
];
