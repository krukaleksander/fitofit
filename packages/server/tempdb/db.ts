export const db = [
  {
    name: 'Aerobik',
    cal: 550,
  },
  {
    name: 'Aerobik w wodzie',
    cal: 580,
  },
  {
    name: 'Badminton',
    cal: 400,
  },
  {
    name: 'Bieg (km przez 3,40 min)',
    cal: 1200,
  },
  {
    name: 'Bieg (km przez 5 min)',
    cal: 1000,
  },
  {
    name: 'Ćwiczenia na siłowni',
    cal: 400,
  },
  {
    name: 'Chodzenie po sklepie',
    cal: 250,
  },
  {
    name: 'Czytanie',
    cal: 25,
  },
  {
    name: 'Czytanie głośno',
    cal: 150,
  },
  {
    name: 'Golf',
    cal: 250,
  },
  {
    name: 'Gotowanie',
    cal: 105,
  },
  {
    name: 'Gra na pianinie 15 min.',
    cal: 120,
  },
  {
    name: 'Gra w kręgle',
    cal: 300,
  },
  {
    name: 'Gra w piłkę nożną',
    cal: 650,
  },
  {
    name: 'Yoga',
    cal: 288,
  },
  {
    name: 'Intensywna gimnastyka',
    cal: 300,
  },
  {
    name: 'Jazda konna',
    cal: 650,
  },
  {
    name: 'Jazda na deskorolce',
    cal: 371,
  },
  {
    name: 'Jazda na nartach biegowych',
    cal: 574,
  },
  {
    name: 'Jazda na nartach wodnych',
    cal: 441,
  },
  {
    name: 'Jazda na nartach zjazdowych',
    cal: 511,
  },
  {
    name: 'Jazda na rolkach',
    cal: 400,
  },
  {
    name: 'Jazda na rowerze (10 km/h)',
    cal: 300,
  },
  {
    name: 'Jazda na rowerze (20 km/h)',
    cal: 600,
  },
  {
    name: 'Jogging',
    cal: 600,
  },
  {
    name: 'Kopanie, grabienie',
    cal: 500,
  },
  {
    name: 'Koszykówka',
    cal: 550,
  },
  {
    name: 'Lekka gimnastyka',
    cal: 210,
  },
  {
    name: 'Lekka praca biurowa',
    cal: 140,
  },
  {
    name: 'Marszobieg',
    cal: 500,
  },
  {
    name: 'Mycie okien',
    cal: 240,
  },
  {
    name: 'Mycie podłogi',
    cal: 250,
  },
  {
    name: 'Odkurzanie',
    cal: 260,
  },
  {
    name: 'Piłowanie drewna',
    cal: 450,
  },
  {
    name: 'Ping-pong',
    cal: 280,
  },
  {
    name: 'Pływanie',
    cal: 400,
  },
  {
    name: 'Praca ekspedientki',
    cal: 280,
  },
  {
    name: 'Praca kamieniarska',
    cal: 400,
  },
  {
    name: 'Praca w ogródku',
    cal: 250,
  },
  {
    name: 'Pranie ręczne',
    cal: 150,
  },
  {
    name: 'Prasowanie',
    cal: 144,
  },
  {
    name: 'Prowadzenie samochodu',
    cal: 126,
  },
  {
    name: 'Robienie na drutach',
    cal: 55,
  },
  {
    name: 'Robienie zakupów',
    cal: 300,
  },
  {
    name: 'Schodzenie ze schodów',
    cal: 364,
  },
  {
    name: 'Ścieranie kurzy',
    cal: 240,
  },
  {
    name: 'Siatkówka',
    cal: 450,
  },
  {
    name: 'Siatkówka plażowa',
    cal: 588,
  },
  {
    name: 'Skakanie na skakance',
    cal: 574,
  },
  {
    name: 'Słanie łóżka',
    cal: 200,
  },
  {
    name: 'Spacer szybkim tempie',
    cal: 300,
  },
  {
    name: 'Spacer w umiarkowanym tempie',
    cal: 200,
  },
  {
    name: 'Sen',
    cal: 62,
  },
  {
    name: 'Śpiew',
    cal: 122,
  },
  {
    name: 'Sprzątanie łazienki',
    cal: 240,
  },
  {
    name: 'Sprzątanie pokoju',
    cal: 180,
  },
  {
    name: 'Squash',
    cal: 497,
  },
  {
    name: 'Stanie na baczność',
    cal: 115,
  },
  {
    name: 'Stanie swobodne',
    cal: 100,
  },
  {
    name: 'Szorowanie podłóg',
    cal: 426,
  },
  {
    name: 'Szybki marsz 6 km/h',
    cal: 150,
  },
  {
    name: 'Szycie na maszynie',
    cal: 135,
  },
  {
    name: 'Szycie ręczne',
    cal: 111,
  },
  {
    name: 'Taniec w dyskotece',
    cal: 500,
  },
  {
    name: 'Tenis',
    cal: 450,
  },
  {
    name: 'Trzepanie dywanów',
    cal: 260,
  },
  {
    name: 'Ubieranie się i rozbieranie',
    cal: 118,
  },
  {
    name: 'Układanie dokumentów',
    cal: 200,
  },
  {
    name: 'Wchodzenie po schodach',
    cal: 1100,
  },
  {
    name: 'Wędkowanie',
    cal: 300,
  },
  {
    name: 'Wiosłowanie',
    cal: 500,
  },
  {
    name: 'Zamiatanie podłogi',
    cal: 100,
  },
  {
    name: 'Zmywanie naczyń',
    cal: 114,
  },
];
