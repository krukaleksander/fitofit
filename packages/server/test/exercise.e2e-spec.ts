import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { AppModule } from '../src/app.module';
import { db } from '../dist/tempdb/db';

describe('Eexercises (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  it('/ (GET)', () => {
    return request(app.getHttpServer())
      .get('/exercises')
      .expect(200)
      .expect(db);
  });
  it('/exercises/new [Post]', async () => {
    return request(app.getHttpServer())
      .post('/exercises/new')
      .send({
        durationInMinutes: 30,
        exerciseID: 2,
        isDone: false,
        name: 'Bieganie',
        start: new Date(),
        userID: 10,
      })
      .expect(201);
  });
  afterAll(async () => {
    await app.close();
  });
});
