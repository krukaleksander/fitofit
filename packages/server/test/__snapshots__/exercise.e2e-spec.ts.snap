// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eexercises (e2e)  returns historical activities exercises/user/activity [Get] 1`] = `
Object {
  "activities": Array [
    Object {
      "activityID": 1,
      "durationInMinutes": 500,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 2,
      "durationInMinutes": 500,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 3,
      "durationInMinutes": 500,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 4,
      "durationInMinutes": 500,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 5,
      "durationInMinutes": 600,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 6,
      "durationInMinutes": 600,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 7,
      "durationInMinutes": 600,
      "exerciseID": 2,
      "isDone": true,
      "name": "Badbinton",
      "start": "2022-04-12T10:49:12.089Z",
      "userID": 10,
    },
    Object {
      "activityID": 8,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T07:57:08.960Z",
      "userID": 10,
    },
    Object {
      "activityID": 9,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T07:58:52.860Z",
      "userID": 10,
    },
    Object {
      "activityID": 10,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:00:33.623Z",
      "userID": 10,
    },
    Object {
      "activityID": 11,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:11:23.418Z",
      "userID": 10,
    },
    Object {
      "activityID": 12,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:11:57.374Z",
      "userID": 10,
    },
    Object {
      "activityID": 13,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:12:13.309Z",
      "userID": 10,
    },
    Object {
      "activityID": 14,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:12:41.759Z",
      "userID": 10,
    },
    Object {
      "activityID": 15,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:28:09.255Z",
      "userID": 10,
    },
    Object {
      "activityID": 16,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:29:50.720Z",
      "userID": 10,
    },
    Object {
      "activityID": 17,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:34:49.956Z",
      "userID": 10,
    },
    Object {
      "activityID": 18,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:36:19.544Z",
      "userID": 10,
    },
    Object {
      "activityID": 19,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:37:46.724Z",
      "userID": 10,
    },
    Object {
      "activityID": 20,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:39:10.693Z",
      "userID": 10,
    },
    Object {
      "activityID": 21,
      "durationInMinutes": 30,
      "exerciseID": 2,
      "isDone": false,
      "name": "Bieganie",
      "start": "2022-04-15T08:41:34.614Z",
      "userID": 10,
    },
  ],
  "caloriesToBurgers": 722,
  "totalCalories": 472993,
  "totalDuration": 4220,
}
`;
